src/main/java/org/reactome/addlinks/referencecreators/KEGGReferenceCreatorHelper.java

```diff
@@ -81,18 +87,28 @@ else if (keggIdentifier.startsWith("ag:"))
 			{
 				targetDB = targetDBID.toString();
 			}
-			// If targetDB is STILL NULL, it means we weren't able to determine which KEGG ReferenceDatabase to use for this keggIdentifier. 
-			// So, we can't add the cross-reference since we don't know which species-specific ReferenceDatabase to use. 
+			// If targetDB is STILL NULL, it means we weren't able to determine which KEGG ReferenceDatabase to use for this keggIdentifier.
+			// So, we can't add the cross-reference since we don't know which species-specific ReferenceDatabase to use.
 			if (targetDB == null)
```

Comment:
This check shouldn't be necessary.  If the `else` block is entered, it's guaranteed that `targetDB` will still be null as it was set on line 70.

Reply:
Might be a legacy from older logic. Will look more closely at this...

Reply:
Wait a minute... what are you looking at? There is no `else` block...

Reply:
The else block starting on line 83.  It contains this line within it.

Reply:
Oh _that_ `else` block...

When then `else` block is entered, `targetDB` will be null. It _could_ get set on line 88 (as a result of the lookup on line 85). But if the kegg prefix does not exist in the database then `targetDB` will still be null _after_ line 88, in which case a new reference database needs to be created.

Reply:
Ah okay - line 88 must have been collapsed when I looked at it so I didn't see it could get set there.  Nevermind about this, then :).